api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts

```diff
@@ -244,4 +248,42 @@ export class SourcingDataImportService {
     }
     return sourcingData;
   }
+
+  async validateFile(
+    filePath: string,
+    filename: string,
+  ): Promise<void | Array<ErrorConstructor>> {
+    let parsedXLSXDataset: SourcingRecordsSheets;
+    try {
+      parsedXLSXDataset = await this.fileService.transformToJson(
+        filePath,
+        SHEETS_MAP,
+      );
+    } catch (error) {
+      throw new ServiceUnavailableException(
+        `File: ${filename} could not have been loaded. Please try again later or contact the administrator`,
+      );
+    }
+
+    /**
+     * Pass random id to create DTOs for validation
+     * @note: If we decide to path DTOs to task instead of path
+     * this has to change to generate valid id
+     */
+    const sourcingLocationGroupId: string =
```

Comment:
this is a hack. any chance we can make the sourcingLocationGroupId an optional argument on the function below (and its dependencies), and validate without it?

Reply:
By model constraints, a SoucingLocation needs to have a SourcingLocationGroupId. But yes, we can avoid this hack passing the 'real' Id of the SourcingLocationGroup we are creating when running an import.

Do you want me to do it in this PR or do you want to see it done in the next PR implementing the 'real' validation?

Reply:
if you promise you won't forget, you can do it in the next PR. but because I don't trust that you won't forget (I trust you, not your memory), add a @todo note here before merging, just in case